sema4ai/vscode-client/src/extension.ts

```diff
@@ -607,10 +607,31 @@ export let globalCachedPythonInfo: InterpreterInfo;
 const langServerMutex: Mutex = new Mutex();
 export let GLOBAL_STATE: undefined | vscode.Memento = undefined;
 
+function checkDeprecatedRobocorpCoreExtension() {
```

Comment:
It might be better to have this as an async function so that the function activation doesn't wait for the extension checks - more of a fire & forget situation.

```suggestion
function async checkDeprecatedRobocorpCoreExtension() {
```

Reply:
Actually the code will run synchronously until the function encounters a promise and the attached .then handler to that promise, the .then handler will be executed asynchronously after the current call stack is complete.

Reply:
I was referring to the check if the VSCode Ext exists. 
If I remember it correctly the `getExtension` function is not async.